x-pack/plugins/maps/server/content_management/maps_storage.ts

```diff
@@ -0,0 +1,312 @@
+      response
+    );
+
+    if (resultError) {
+      throw Boom.badRequest(`Invalid response. ${resultError.message}`);
+    }
+
+    return value;
+  }
+
+  async bulkGet(): Promise<never> {
+    // Not implemented. Maps does not use bulkGet
+    throw new Error(`[bulkGet] has not been implemented. See MapsStorage class.`);
+  }
+
+  async create(
+    ctx: StorageContext,
+    data: MapCreateIn['data'],
+    options: CreateOptions
+  ): Promise<MapCreateOut> {
+    const {
+      utils: { getTransforms },
+      version: { request: requestVersion },
+    } = ctx;
+    const transforms = getTransforms(cmServicesDefinition, requestVersion);
+
+    // Validate input (data & options) & UP transform them to the latest version
+    const { value: dataToLatest, error: dataError } = transforms.create.in.data.up<
+      MapAttributes,
+      MapAttributes
+    >(data);
+    if (dataError) {
+      throw Boom.badRequest(`Invalid data. ${dataError.message}`);
+    }
+
+    const { value: optionsToLatest, error: optionsError } = transforms.create.in.options.up<
+      CreateOptions,
+      CreateOptions
+    >(options);
+    if (optionsError) {
+      throw Boom.badRequest(`Invalid options. ${optionsError.message}`);
+    }
+
+    // Save data in DB
+    const soClient = await savedObjectClientFromRequest(ctx);
+    const savedObject = await soClient.create<MapAttributes>(
+      SO_TYPE,
+      dataToLatest,
+      optionsToLatest
+    );
+
+    // Validate DB response and DOWN transform to the request version
+    const { value, error: resultError } = transforms.create.out.result.down<
+      MapCreateOut,
+      MapCreateOut
+    >({
+      item: savedObjectToMapItem(savedObject, false),
+    });
+
+    if (resultError) {
+      throw Boom.badRequest(`Invalid response. ${resultError.message}`);
+    }
+
+    return value;
+  }
+
+  async update(
+    ctx: StorageContext,
+    id: string,
+    data: MapUpdateIn['data'],
+    options: UpdateOptions
+  ): Promise<MapUpdateOut> {
+    const {
+      utils: { getTransforms },
+      version: { request: requestVersion },
+    } = ctx;
+    const transforms = getTransforms(cmServicesDefinition, requestVersion);
+
+    // Validate input (data & options) & UP transform them to the latest version
+    const { value: dataToLatest, error: dataError } = transforms.update.in.data.up<
+      MapAttributes,
+      MapAttributes
+    >(data);
+    if (dataError) {
+      throw Boom.badRequest(`Invalid data. ${dataError.message}`);
+    }
+
+    const { value: optionsToLatest, error: optionsError } = transforms.update.in.options.up<
+      CreateOptions,
+      CreateOptions
+    >(options);
+    if (optionsError) {
+      throw Boom.badRequest(`Invalid options. ${optionsError.message}`);
+    }
+
+    // Save data in DB
+    const soClient = await savedObjectClientFromRequest(ctx);
+    const partialSavedObject = await soClient.update<MapAttributes>(
```

Comment:
Why does this work with partial saved objects? Is there something about the SO client that makes this necessary?